### DISCUSSION TITLE
API PIX - CONSULTAR CHAVE PIX

### DISCUSSION THREAD

**Discussion by thalissomvinicius** [bacen/pix-api#588] [General]
Olá, alguém pode me ajudar.
Estou com um desafio de fazer um projeto que verifique a existencia de uma determianda chave pix, se ela é ou não já cadastrada.
Alguém pode me dar uma luz, ou me mandar um link, api, algo do tipo por gentiliza?

**Comment by rafaelchagasb** [comment]
Hoje não existe esse recurso na Api Pix.

O Bacen deixa as instituições participantes do Pix fazerem essa consulta usando a API do DICT. Mesmo assim a API tem um limite bem pequeno de requisições(Rate Limit).

https://www.bcb.gov.br/content/estabilidadefinanceira/pix/API-DICT-2.0.1.html#tag/Key/operation/checkKeys

**Reply by thalissomvinicius** [reply]
Com esse limite, fica dificil fazer as requisições que eu quero kkkk.
Estou apanhando para fazer a integração com api, só da erro!!!!

**Reply by rubenskuhl** [reply]
Mas está dando erro em qual PSP ? Você não sendo PSP não tem acesso aos sistemas do BACEN.

**Reply by rubenskuhl** [reply]
> Se tiver precisando de api consulta pix:, Chama no whatsapp: 81989769960

O @ninrod e @thiagolvlsantos tem interesse

**Comment by rubenskuhl** [comment]
> Olá, alguém pode me ajudar. Estou com um desafio de fazer um projeto que verifique a existencia de uma determianda chave pix, se ela é ou não já cadastrada. Alguém pode me dar uma lua, ou me mandar um link, api, algo do tipo por gentiliza?

Mande um Pix de 1 centavo, se voltar erro, a chave não existe. 
A falta desse caso de uso não é uma omissão, é uma decisão pensada do Banco Central. Não tem luz ou link, exceto você repensar esse uso.

**Reply by furlongsolotov** [reply]
KKKKKKKKKKKKKKKKK

**Comment by eletroswing** [comment]
Olá @thalissomvinicius, eu estive desenvolvendo um simulador do bacen ultimamente antes de meu computador quebrar, e implementei a parte do dict que cuida das chave pix, se for necessário você pode usá-lo para simular a comunicação real com o bacen. Em breve o sistema contará com o simulador do spi também. Segue o link
https://github.com/eletroswing/bacen-simulator

É a penas um simulador e não a, e não deve haver dados de pessoas reais dentro do sistema

**Comment by aeceloto** [comment]
É possivel verificar a existência de chaves, sem consultar os detalhes
através da API do DICT https://dict-np-h.pi.rsfn.net.br:16532/api-np/v1
/keys/check.


Está API ela retorna somente se a chave é válida ou não.

Le mer. 3 avr. 2024, 13:41, Thalissom Vinicius ***@***.***> a
écrit :

**Reply by TheSeth7** [reply]
como usa

**Reply by rubenskuhl** [reply]
> É possivel verificar a existência de chaves, sem consultar os detalhes através da API do DICT https://dict-np-h.pi.rsfn.net.br:16532/api-np/v1 /keys/check. Está API ela retorna somente se a chave é válida ou não. Le mer. 3 avr. 2024, 13:41, Thalissom Vinicius ***@***.***> a écrit :
> […](#)
> Olá, alguém pode me ajudar. Estou com um desafio de fazer um projeto que verifique a existencia de uma determianda chave pix, se ela é ou não já cadastrada. Alguém pode me dar uma lua, ou me mandar um link, api, algo do tipo por gentiliza? — Reply to this email directly, view it on GitHub <#588>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGTMB52HR3Q5YGDJ5FWTHSTY3P2JNAVCNFSM6AAAAABFVHUVTWVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGQ2TIOJRGM> . You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Como está implícito na URL que você postou (.rsfn.net.br), isso só é acessível via rede fechada mantida pelo BACEN, a RSFN (Rede do Sistema Financeiro Nacional). E que mesmo alguém que tenha acesso na RSFN, só recebe credenciais para APIs do BACEN para o Pix se for participante do Pix. 

E especificamente dessa chamada, ela só informa existência de chaves celular e e-mail... ela só está autorizada para consulta pelos participantes do Pix cruzando o address book dos correntistas desses participantes com a existência de chaves no DICT, e não qualquer outro caso de uso. 

A saída prática que existe continua sendo enviar um valor para uma possível chave, e ver se acontece devolução ou o envio conclui com sucesso.

**Comment by TheSeth7** [comment]
voce conseguiu mano?? se sim como

**Comment by merloch22** [comment]


**Reply by rubenskuhl** [reply]
Isso não está proibido desde o último 02.09 ?

**Comment by lmarqs** [comment]
Olá,
Gostaria muito que existisse essa funcionalidade.
Tenho clientes que cadastram seu dados para recebimento em nossa plataforma e gostaria de validar se a chave pix cadastrada tem correlação com o documento da pessoa.
Assim conseguimos validar se a conta não foi invadida e tentaram colocar uma chave de outra pessoa.
Vi que existem algumas APIs que fazem isso e eles devolvem o documento da pessoa mascarado, a hash do ducumento, o nome da pessoa, a instituição, e o numero da conta.
É a mesma coisa quando vamos pagar via aplicativo do banco. Ao digitar a chave o banco consulta no dict e mostra para nós essas informações. Assim conseguimos cruzar com o documento da pessoa (via hash) e também na hora do pagamento verificar se a chave pix não foi transferida para outra conta alertando nosso time para a possibilidade de necessitar algum tipo de segunda verificação para nosso time antes de pagar naquela chave.
Nossa volumetria é extremamente pequena, e com certeza esta bem abaixo do rate limit da api dict.
Obrigado.

**Reply by rubenskuhl** [reply]
Mesmo APIs que corretamente não deixam consultar o DICT podem restringir o envio de Pix para determinada chave por CPF/CNPJ. É o caso da Efí, aonde você pode passar chave Pix e CPF/CNPJ, e a transação não é concluída se o CPF/CNPJ não bater com a chave. Então ter uma API de consulta não é o único jeito de mitigar o risco que você apontou.

**Reply by renatofrota** [reply]
Outra solução mais simples e universal é simplesmente exigir que a chave do recebedor seja no formato CPF. Pode ser inconveniente mas é a forma mais certeira de determinar o recebedor do pix.

**Reply by lmarqs** [reply]
> Mesmo APIs que corretamente não deixam consultar o DICT podem restringir o envio de Pix para determinada chave por CPF/CNPJ. É o caso da Efí, aonde você pode passar chave Pix e CPF/CNPJ, e a transação não é concluída se o CPF/CNPJ não bater com a chave. Então ter uma API de consulta não é o único jeito de mitigar o risco que você apontou.

Sim, avaliamos também essa alternativa.
No caso precisaríamos fazer um PIX de R$ 0,01 para confirmar que a chave pertence a pessoa. Foi uma das alternativas levantadas.
Outra forma seria na hora do cadastro termos uma pessoa aqui da empresa que abrisse o celular e validasse via aplicativo do banco.
O importante para nosso cenário é validar principalmente no cadastro, antes das transações.

Obrigado.

**Reply by rubenskuhl** [reply]
> > Mesmo APIs que corretamente não deixam consultar o DICT podem restringir o envio de Pix para determinada chave por CPF/CNPJ. É o caso da Efí, aonde você pode passar chave Pix e CPF/CNPJ, e a transação não é concluída se o CPF/CNPJ não bater com a chave. Então ter uma API de consulta não é o único jeito de mitigar o risco que você apontou.
> 
> Sim, avaliamos também essa alternativa. No caso precisaríamos fazer um PIX de R$ 0,01 para confirmar que a chave pertence a pessoa. Foi uma das alternativas levantadas. Outra forma seria na hora do cadastro termos uma pessoa aqui da empresa que abrisse o celular e validasse via aplicativo do banco. O importante para nosso cenário é validar principalmente no cadastro, antes das transações.


Então, esse teste (que custa algo como 51 centavos entre tarifa e valor enviado) talvez não seja necessário. Se na primeira transação para valer der problema, avisa o cliente para ele corrigir... você pode fazer validação sintática da chave e algorítmica do CPF que vai cobrir bem erros de digitação. E ainda influenciar na UI para que seja mais simples escolher a chave CPF, e no agregado tornar isso um não-problema.

**Reply by lmarqs** [reply]
Dado nosso fluxo de operação o ideal é que seja validado na hora do cadastro.
Essa validação é principalmente para chaves pix email, telefone e aleatoria.

De qualquer forma ja estamos entrando em contato com fornecedores que proveem esse funcionalidade:
https://docs.bankly.com.br/docs/pix-consulta-por-chaves
https://developers.bitcapital.com.br/docs/consultar-uma-chave
https://developers.celcoin.com.br/reference/retornar-informacoes-do-dict-utilizando-uma-chave-cadastrada-pix-indireto

Deixo a sugestão pois pode ser util para outras pessoas.

**Reply by rubenskuhl** [reply]
> Dado nosso fluxo de operação o ideal é que seja validado na hora do cadastro. Essa validação é principalmente para chaves pix email, telefone e aleatoria.
> 
> De qualquer forma ja estamos entrando em contato com fornecedores que proveem esse funcionalidade: https://docs.bankly.com.br/docs/pix-consulta-por-chaves https://developers.bitcapital.com.br/docs/consultar-uma-chave https://developers.celcoin.com.br/reference/retornar-informacoes-do-dict-utilizando-uma-chave-cadastrada-pix-indireto
> 
> Deixo a sugestão pois pode ser util para outras pessoas.

Você acabou de colocar esses 3 prestadores como violando normas do Banco Central, num github do Banco Central... @ninrod @aldenio não vão ter muita opção.

**Reply by lmarqs** [reply]
> 
> Você acabou de colocar esses 3 prestadores como violando normas do Banco Central, num github do Banco Central... @ninrod @aldenio não vão ter muita opção.

Desconheço qualquer irregularidade da situação. Apenas compartilhei resultados de uma busca rápida do Google.

Mas já deu para entender que existe uma grande resistência a essa sugestão de funcionalidade.

Encerro aqui minha participação nessa discussão.

Obrigado

**Reply by rubenskuhl** [reply]
> > Você acabou de colocar esses 3 prestadores como violando normas do Banco Central, num github do Banco Central... @ninrod @aldenio não vão ter muita opção.
> 
> Desconheço qualquer irregularidade da situação. Apenas compartilhei resultados de uma busca rápida do Google.

https://www.bcb.gov.br/content/estabilidadefinanceira/pix/Forum_Pix_Plenaria/20241205-Forum_Pix.pdf


<img width="947" alt="Captura de Tela 2024-12-20 às 01 52 45" src="https://github.com/user-attachments/assets/9c771a21-8af0-4a56-ad07-e20aac018602" />
